reana/reana_dev/release.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2020 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""`reana-dev`'s release commands."""
+
+import sys
+from time import sleep
+
+import click
+
+from reana.reana_dev.git import (
+    git_clean,
+    git_is_current_version_tagged,
+    is_last_commit_release_commit,
+)
+from reana.reana_dev.utils import (
+    fetch_latest_pypi_version,
+    get_current_component_version_from_source_files,
+    run_command,
+    select_components,
+)
+
+
+@click.group()
+def release_commands():
+    """Release commands group."""
+
+
+@click.option(
+    "--component",
+    "-c",
+    required=True,
+    multiple=True,
+    help="Which components? [name|CLUSTER]",
+)
+@click.option(
+    "--timeout",
+    required=True,
+    type=int,
+    default=90,
+    help="How many seconds should we wait to confirm successful PyPI release?",
+)
+@release_commands.command(name="release-pypi")
+@click.pass_context
+def release_pypi(ctx, component, timeout):  # noqa: D301
+    """Release a component on pypi.org.
+
+    \b
+    :param components: The option ``component`` can be repeated. The value may
+                       consist of:
+                         * (1) standard component name such as
+                               'reana-workflow-controller';
+                         * (2) short component name such as 'r-w-controller';
+                         * (3) special value '.' indicating component of the
+                               current working directory;
+                         * (4) special value 'CLUSTER' that will expand to
+                               cover all REANA cluster components [default];
+                         * (5) special value 'CLIENT' that will expand to
+                               cover all REANA client components;
+                         * (6) special value 'DEMO' that will expand
+                               to include several runable REANA demo examples;
+                         * (7) special value 'ALL' that will expand to include
+                               all REANA repositories.
+    :type component: str
+    """
+    components = select_components(component)
```

Comment:
Compontent parsing produces:

```
~r/reana pr-376 reana $ reana-dev release-pypi -c r-commons
[2020-08-12T16:48:08] : Ignoring unknown component r.
[2020-08-12T16:48:08] : Ignoring unknown component e.
[2020-08-12T16:48:08] : Ignoring unknown component a.
[2020-08-12T16:48:08] : Ignoring unknown component n.
[2020-08-12T16:48:08] : Ignoring unknown component a.
[2020-08-12T16:48:08] : Ignoring unknown component -.
[2020-08-12T16:48:08] : Ignoring unknown component c.
[2020-08-12T16:48:08] : Ignoring unknown component o.
[2020-08-12T16:48:08] : Ignoring unknown component m.
[2020-08-12T16:48:08] : Ignoring unknown component m.
[2020-08-12T16:48:08] : Ignoring unknown component o.
[2020-08-12T16:48:08] : Ignoring unknown component n.
[2020-08-12T16:48:08] : Ignoring unknown component s.
```

Afterwards it well continues with `reana-commons`.